ush/wafs_grib2_0p25_blending.sh

```diff
@@ -0,0 +1,145 @@
+    # Three(3) unblended UK files for each cycle+fhour: icing, turb, cb
+    ukfiles=$(find "${COMINuk}" -name "egrr_wafshzds_unblended_*_0p25_${PDY:0:4}-${PDY:4:2}-${PDY:6:2}T${cyc}:00Z_t${fhr}.grib2" | wc -l)
+    if ((ukfiles >= 3)); then
+        echo "Found all 3 UK WAFS GRIB2 files, continue ..."
+        break
+    fi
+
+    if ((ic == SLEEP_LOOP_MAX)); then
+        products="cb ice turb"
+        for prod in ${products}; do
+            ukfile="egrr_wafshzds_unblended_${prod}_0p25_${PDY:0:4}-${PDY:4:2}-${PDY:6:2}T${cyc}:00Z_t${fhr}.grib2"
+            if [[ ! -f "${COMINuk}/${ukfile}" ]]; then
+                echo "WARNING: UK WAFS GRIB2 file '${ukfile}' not found after waiting over ${SLEEP_TIME} seconds"
+                echo "Missing ${COMINuk}/${ukfile}" >> ../missing_uk_files.$fhr
+            fi
+        done
+        echo "WARNING: UK WAFS GRIB2 unblended data is not completely available, no blending"
+        MISSING_UK_WAFS="YES"
+        break
+    else
+        sleep "${SLEEP_INT}"
+    fi
+done
+
+##########################
+# look for US WAFS data.
+##########################
+MISSING_US_WAFS="NO"
+
+if [[ ! -f "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2.idx" ]]; then
+    # Blending ecflow run is triggered by the finishing of the upstream JWAFS_GRIB2_0P25 f048
+    #  - In real time ecflow, blending starts at T+4:30, unblended upstream products are all available.
+    #    It doesn't need to sleep/wait
+    #  - In non-real time ecflow, blending has no time tigger, unblended f048 is not guaranteed to be the last finished one.
+    #    It needs to sleep/wait. All upstream jobs can be finished within 60 seconds when f048 is finished.
+    #
+    # Standalone blending doesn't need to sleep/wait since unblended upstream grib2_0p25 runs standalone first.
+    sleep 60
+    if [[ ! -f "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2.idx" ]]; then
+	echo "WARNING: missing US unblended data - ${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2"
+	echo "Missing ${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2" > ../missing_us_files.$fhr
+	MISSING_US_WAFS="YES"
+    fi
+fi
+
+##########################
+# Blending or unblended
+##########################
+if [[ "${MISSING_UK_WAFS}" == "YES" ]] && [[ "${MISSING_US_WAFS}" == "YES" ]]; then
+    cat ../missing_uk_files.$fhr ../missing_us_files.$fhr > ../no_blending_files.$fhr
+    rm ../missing_uk_files.$fhr ../missing_us_files.$fhr
+elif [[ "${MISSING_UK_WAFS}" == "YES" ]]; then
+    echo "turning back on dbn alert for unblended US WAFS product"
+    # Avoid duplicate dbn_alert of unblended grib2 file which was done in the upstream grib2_0p25 job, fix bugzilla 1226
+    # "${DBNROOT}/bin/dbn_alert" MODEL WAFS_0P25_UBL_GB2 "${job}" "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2"
+    "${DBNROOT}/bin/dbn_alert" MODEL WAFS_0P25_UBL_GB2_WIDX "${job}" "${COMINus}/WAFS_0p25_unblended_${PDY}${cyc}f${fhr}.grib2.idx"
```

Comment:
I suggest this section be right after line 63 where it has determined UK data is missing, and send the alert right there.
See similar comment for US below.

Reply:
If line 98-101 is right after 63, we don't know whether the unblended US data is present yet

Reply:
a DBNAlert is not an abort.  The US data will be alerted when that is encountered.
If both are missing, then there are 2 alerts sent.
Moving the alert statements do not preclude from other not being present/absent

Reply:
Will dbn_alert create an error message if the file to be alerted is not present?

Reply:
It is an alert, like an echo.

Reply:
If both UK and US data are missing, moving these lines will make dbn_alert some file which doesn't exist.

Reply:
That is not how dbn_alert works.  Test with `fakedbn` to find out.

Reply:
I just checked, dbn_alert needs 4 arguments:
/apps/ops/prod/nco/core/prod_util.v2.0.14/fakedbn/bin/dbn_alert
/apps/ops/prod/nco/core/prod_util.v2.0.14/para_dbn/bin/dbn_alert